robottelo/cli/factory.py

```diff
@@ -179,6 +180,88 @@ def make_architecture(options=None):
+        --attach-stdin ATTACH_STDIN               One of true/false, yes/no,
+                                                  1/0.
+        --attach-stdout ATTACH_STDOUT             One of true/false, yes/no,
+                                                  1/0.
+        --capsule CAPSULE_NAME                    Name to search by
+        --capsule-id CAPSULE_ID                   Id of the capsule
+        --command COMMAND
+        --compute-resource COMPUTE_RESOURCE_NAME  Compute resource name
+        --compute-resource-id COMPUTE_RESOURCE_ID
+        --cpu-sets CPU_SETS
+        --cpu-shares CPU_SHARES
+        --entrypoint ENTRYPOINT
+        --location-ids LOCATION_IDS               REPLACE locations with given
+        ids. Comma separated list of values.
+        --locations LOCATION_NAMES                Comma separated list of
+                                                  values.
+        --memory MEMORY
+        --name NAME
+        --organization-ids ORGANIZATION_IDS       REPLACE organizations with
+                                                  given ids.  Comma separated
+                                                  list of values.
+        --organizations ORGANIZATION_NAMES        Comma separated list of
+                                                  values.
+        --registry-id REGISTRY_ID                 Registry this container will
+                                                  have to use to get the image
+        --repository-name REPOSITORY_NAME         Name of the repository to use
+                                                  to create the container. e.g:
+                                                  centos
+        --tag TAG                                 Tag to use to create the
+                                                  container. e.g: latest
+        --tty TTY                                 One of true/false, yes/no,
+                                                  1/0.
+
+    """
+    # Organization ID is a required field.
+    if (not options or (
+            u'compute-resource' not in options and
+            u'compute-resource-id' not in options
+    )):
+        raise CLIFactoryError(
+            'Please provide at least compute-resource or compute-resource-id '
+            'options.'
+        )
+
+    args = {
+        u'attach-stderr': None,
+        u'attach-stdin': None,
+        u'attach-stdout': None,
+        u'capsule': None,
+        u'capsule-id': None,
+        u'command': 'top',
+        u'compute-resource': None,
+        u'compute-resource-id': None,
+        u'cpu-sets': None,
+        u'cpu-shares': None,
+        u'entrypoint': None,
+        u'location-ids': None,
+        u'locations': None,
+        u'memory': None,
+        u'name': gen_string('utf8'),
+        u'organization-ids': None,
+        u'organizations': None,
+        u'registry-id': None,
+        u'repository-name': 'busybox',
```

Comment:
This is not clear on the current version of the help, check https://github.com/SatelliteQE/robottelo/pull/2016/files#diff-ba0ea34671620ba50fb287b4c4913463R73 and you will see that it suggests the busybox. To pass a katello repo instead need to use the full name like showed here https://github.com/theforeman/foreman-docker/pull/93#issue-59835151.

As the feature is not completed yet, I'm sending the current work I was able to do based on my assumptions, this will need to be reviewed once we finally get the feature working.